tower-balance/src/choose/mod.rs

```diff
@@ -0,0 +1,23 @@
+use ordermap::OrderMap;
+
+use Load;
+
+mod p2c;
+mod round_robin;
+
+pub use self::p2c::PowerOfTwoChoices;
+pub use self::round_robin::RoundRobin;
+
+/// A strategy for choosing nodes.
+pub trait Choose {
+    type Metric: PartialOrd;
+
+    /// Returns the index of a ready endpoint.
+    ///
+    /// ## Panics
+    ///
+    /// If `nodes` has fewer than 2 entries.
+    fn call<K, L>(&mut self, nodes: &OrderMap<K, L>) -> usize
```

Comment:
> Don't expose OrderMap directly, instead wrap it in a new type.

Yeah this is a good idea.

> Ideally, the K wouldn't be exposed. Unfortunately, that would be a bit annoying (though, possible) to hide.

Have an example handy?

> Having generics on the fn means that it can't be used as a trait object.

What does this mean, practically?